service/controller/v3/key/key.go

```diff
@@ -57,6 +58,13 @@ func IsDeleted(customObject v1alpha1.FlannelConfig) bool {
 	return customObject.GetDeletionTimestamp() != nil
 }
 
+// MaxUnavailable is used for the Kubernetes update strategy. We want only one
+// pod at a time to be unavailable during updates.
```

Comment:
That's nice. We should think more about high availability in the k8s resources we generate in the operators.

Reply:
Have you checked how long this takes with `1` max unavailable? Just out of curiosity, because we are usually too aggressive but this is very safe.